File: WebApp/WebApp/Models/UserTypeDiscount.cs
﻿using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApp.Models
{
    public class UserTypeDiscount
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [ForeignKey("UserType")]
        public int UserTypeId { get; set; }
        public virtual UserType UserType { get; set; }

        [Required]
        public float Discount { get; set; }

        public virtual ICollection<PriceListItem> PriceListItems { get; set; }

        public UserTypeDiscount()
        {
            this.PriceListItems = new HashSet<PriceListItem>();
        }

    }
}

File: NodeJsApp/src/models/userTypeDiscount.js
import { Schema, model } from 'mongoose';

const userTypeDiscountSchema = new Schema({
    discount: {
        type: Schema.Types.Decimal128,
        required: true
    },
    userType: {
        type: Schema.Types.ObjectId,
        ref: 'UserType',
        required: true
    },
    priceListItems: [
        {
            type: Schema.Types.ObjectId,
            ref: 'PriceListItem'
        }
    ]
});

export default model('UserTypeDiscount', userTypeDiscountSchema);